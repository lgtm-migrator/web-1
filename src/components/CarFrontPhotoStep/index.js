import React from "react"

import Paper from "@material-ui/core/Paper"
import Fab from "@material-ui/core/Fab"

import AddAPhotoIcon from "@material-ui/icons/AddAPhoto"
import Alert from "@material-ui/lab/Alert"
import AlertTitle from "@material-ui/lab/AlertTitle"

import {
  Fieldset,
  Figure,
  FigureCaption,
  Form,
  H2,
  Image,
  Paragraph,
  PhotoUploadField,
} from "muy"

import Fade from "@material-ui/core/Fade"
import Typography from "@material-ui/core/Typography"

const CarFrontPhotoStep = ({
  isCarFrontPhotoValid,
  onChange,
  onSubmit,
  photoPreviewURL,
}) => (
  <Form onSubmit={onSubmit}>
    <Fieldset>
      <H2 gutterBottom={true}>
        Tire uma foto da&nbsp;frente do&nbsp;carro, com&nbsp;a&nbsp;placa
        visível
      </H2>
      <Paragraph variant="body2">
        A&nbsp;foto precisa mostrar o&nbsp;painel do&nbsp;veículo e&nbsp;provar
        que&nbsp;não&nbsp;há uma credencial visível.
      </Paragraph>
      {!isCarFrontPhotoValid && false && (
        <Alert severity="warning" variant="outlined">
          <AlertTitle>
            Por favor, envie uma foto da parte da frente do veículo.
          </AlertTitle>
        </Alert>
      )}
      <Paragraph>
        <Fab
          aria-label="Adicionar foto da frente"
          color="primary"
          component="label"
          htmlFor="carFrontImageUploadField"
          role={null}
          tabIndex={0}
          variant="extended"
        >
          <AddAPhotoIcon />
          &nbsp;
          {photoPreviewURL ? "Trocar foto" : "Adicionar foto"}
        </Fab>
        <Typography component={"span"} variant={"srOnly"}>
          <PhotoUploadField
            id="carFrontImageUploadField"
            data-test-id="carFrontImageUploadField"
            onChange={onChange}
            name="carFrontImage"
            required={true}
            tabIndex={-1}
            capture={null}
          />
        </Typography>
      </Paragraph>
      <Fade in={Boolean(photoPreviewURL)}>
        <Paper>
          <Figure>
            <Typography component={"div"} variant={"srOnly"}>
              <FigureCaption>Prévia da frente do veículo</FigureCaption>
            </Typography>
            <Image
              alt="Prévia da frente do veículo"
              loading="lazy"
              src={photoPreviewURL}
            />
          </Figure>
        </Paper>
      </Fade>
    </Fieldset>
  </Form>
)

export default CarFrontPhotoStep
