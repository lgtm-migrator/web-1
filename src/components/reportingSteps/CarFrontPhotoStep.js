import React from "react"
import FileInput from "../ui/FileInput"
import Typography from "@material-ui/core/Typography"
import Alert from "@material-ui/lab/Alert"
import AlertTitle from "@material-ui/lab/AlertTitle"
import Figure from "../ui/Figure"
import Fab from "@material-ui/core/Fab"
import AddAPhotoIcon from "@material-ui/icons/AddAPhoto"
import Image from "../ui/Image"

const CarFrontPhotoStep = ({
  isCarFrontPhotoValid,
  onChange,
  onSubmit,
  photoPreviewURL,
}) => (
  <form onSubmit={onSubmit}>
    <Typography component="h2" variant="h2">
      Tire uma foto da frente do carro, com a placa visível
    </Typography>
    <Typography paragraph={true} variant="body2">
      A foto precisa mostrar o painel do veículo e provar que não há uma
      credencial visível.
    </Typography>
    {!isCarFrontPhotoValid && false && (
      <Alert severity="warning" variant="outlined">
        <AlertTitle>
          Por favor, envie uma foto da parte da frente do veículo.
        </AlertTitle>
      </Alert>
    )}
    {photoPreviewURL && (
      <Figure>
        <figcaption>Prévia da parte da frente do veículo</figcaption>
        <Image
          alt="Prévia da frente do veículo"
          loading="lazy"
          src={photoPreviewURL}
        />
      </Figure>
    )}
    <Typography paragraph={true}>
      <Fab
        aria-label="adicionar foto da frente"
        color="primary"
        component="label"
        htmlFor="carFrontImageUploadField"
        tabIndex={0}
        variant="extended"
      >
        <AddAPhotoIcon />
        Adicionar foto
      </Fab>
      <Typography variant="srOnly">
        <FileInput
          id="carFrontImageUploadField"
          accept="image/*"
          data-test-id="carFrontImageUploadField"
          onChange={onChange}
          name="carFrontImage"
          required={true}
          tabIndex={-1}
        />
      </Typography>
    </Typography>
  </form>
)

export default CarFrontPhotoStep
