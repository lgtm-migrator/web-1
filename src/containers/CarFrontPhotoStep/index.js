import React from "react"
// import CarFrontPhotoStep from "../../components/CarFrontPhotoStep"
import useGlobal from "../../hooks/useGlobal"
import TakeCarFrontPhotoStep from "../../components/TakeCarFrontPhotoStep"

export default ({ onSubmit, ...props }) => {
  const [globalState, globalActions] = useGlobal()

  const handleFileUpload = (event) => {
    event.persist()
    globalActions.setLoading(true)

    if (typeof event.target.files[0] === "undefined") {
      globalActions.setLoading(false)
      globalActions.setCurrentParkingReportingData({
        ...globalState.currentParkingReportingData,
        carFrontPhotoPreviewUrl: null,
      })
      return false
    }

    const file = event.target.files[0]
    globalActions.setCurrentParkingReportingData({
      ...globalState.currentParkingReportingData,
      carFrontPhotoPreviewUrl: URL.createObjectURL(file),
    })

    globalActions.setLoading(false)
  }
  const handleSubmit = (event) => {
    event.persist()
    event.preventDefault()
    globalActions.setLoading(true)
    onSubmit()
  }
  const { carFrontPhotoPreviewUrl } = globalState.currentParkingReportingData

  return (
    <TakeCarFrontPhotoStep
      onChange={handleFileUpload}
      onSubmit={handleSubmit}
      photoPreviewURL={carFrontPhotoPreviewUrl}
      {...props}
    />
  )
}
