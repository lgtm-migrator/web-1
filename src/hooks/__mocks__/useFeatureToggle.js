export const useFeatureToggleMock = jest.fn()
