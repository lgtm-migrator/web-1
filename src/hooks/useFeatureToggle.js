import { useDebugValue, useEffect, useState } from "react"

const useFeatureToggle = (key) => {
  const [value, setValue] = useState(null)

  const valueSanitizer = (value) => {
    if (typeof value === "undefined") {
      return false
    }

    switch (value) {
      case 1:
      case "1":
      case "true":
      case "TRUE":
        return true
      default:
        return false
    }
  }

  useDebugValue(value)

  useEffect(() => {
    console.log("Value is", value)
  }, [value])

  useEffect(() => {
    console.info("Getting value from env variable", key)

    const keys = {
      GET_USER_MEDIA: process.env.GATSBY_TOGGLE_GET_USER_MEDIA,
      PARKING_CHECK: process.env.GATSBY_TOGGLE_PARKING_CHECK,
      PARKING_REPORT: process.env.GATSBY_TOGGLE_PARKING_REPORT,
    }

    if (keys[key]) {
      const envValue = valueSanitizer(keys[key])
      console.info("Env value is", envValue)
      setValue(envValue)
    }
  }, [key])

  const localStorageKey = `TOGGLE_${key}`

  useEffect(() => {
    console.info("Getting value from local storage")

    if (localStorage.getItem(localStorageKey)) {
      const localValue = valueSanitizer(localStorage.getItem(localStorageKey))
      console.info("Local storage value is", localValue)
      setValue(localValue)
    }
  }, [localStorageKey])

  useEffect(() => {
    const listener = (event) => {
      if (event.key === localStorageKey) {
        const localChangedValue = valueSanitizer(event.newValue)
        console.info("Local storage changed value is", localChangedValue)
        setValue(localChangedValue)
      }
    }
    window.addEventListener("storage", listener)
    return () => window.removeEventListener("storage", listener)
  }, [localStorageKey])

  return [value]
}

export default useFeatureToggle
