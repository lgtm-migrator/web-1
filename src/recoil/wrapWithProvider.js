import React from "react"
import { RecoilRoot } from "recoil"

// eslint-disable-next-line react/display-name,react/prop-types
export const wrapWithProvider = ({ element }) => (
  <RecoilRoot>{element}</RecoilRoot>
)
