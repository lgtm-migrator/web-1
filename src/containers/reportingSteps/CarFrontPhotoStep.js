import React from "react"
import CarFrontPhotoStep from "../../components/reportingSteps/CarFrontPhotoStep"
import useGlobal from "../../hooks/useGlobal"

export default (props) => {
  const [globalState, globalActions] = useGlobal()

  const handleFileUpload = (event) => {
    event.persist()
    globalActions.setLoading(true)

    if (typeof event.target.files[0] === "undefined") {
      globalActions.setLoading(false)
      globalActions.setCurrentParkingReportingData({
        ...globalState.currentParkingReportingData,
        carFrontPhotoPreviewUrl: null,
      })
      return false
    }

    const file = event.target.files[0]
    globalActions.setCurrentParkingReportingData({
      ...globalState.currentParkingReportingData,
      carFrontPhotoPreviewUrl: URL.createObjectURL(file),
    })
    console.log(globalState.currentParkingReportingData)

    globalActions.setLoading(false)
  }
  const handleSubmit = (event) => {
    event.persist()
    event.preventDefault()
    globalActions.setLoading(true)
  }
  const { carFrontPhotoPreviewUrl } = globalState.currentParkingReportingData

  return (
    <CarFrontPhotoStep
      onChange={handleFileUpload}
      onSubmit={handleSubmit}
      photoPreviewURL={carFrontPhotoPreviewUrl}
      {...props}
    />
  )
}
