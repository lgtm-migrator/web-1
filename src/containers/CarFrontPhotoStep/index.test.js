import React from "react"
import renderer from "react-test-renderer"
import { RecoilRoot } from "recoil"

import CarFrontPhotoStep from "."
import useFeatureToggle from "./../../hooks/useFeatureToggle"

jest.mock("./../../hooks/useFeatureToggle")
jest.mock("./../../hooks/useLoadingState")
jest.mock("./../../hooks/useParkingReportState")

describe("CarFrontPhotoStep", () => {
  let props

  beforeEach(() => {
    props = {
      onSubmit: () => {},
    }
    useFeatureToggle.mockReturnValue([false])
  })

  it("should render correctly", () => {
    const tree = renderer
      .create(
        <RecoilRoot>
          <CarFrontPhotoStep {...props} />
        </RecoilRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
