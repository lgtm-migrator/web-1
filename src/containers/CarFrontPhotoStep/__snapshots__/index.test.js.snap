// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarFrontPhotoStep should render correctly 1`] = `
<form
  className="MuiBox-root MuiBox-root-2"
  data-name="form"
  id="form"
  name="form"
  onSubmit={[Function]}
>
  <fieldset
    className="MuiBox-root MuiBox-root-6 makeStyles-root-5"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
    >
      Tire uma foto da frente do carro, com a placa visível
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-paragraph"
    >
      A foto precisa mostrar o painel do veículo e provar que não há uma credencial visível.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      <label
        aria-disabled={false}
        aria-label="Adicionar foto da frente"
        className="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-primary"
        htmlFor="carFrontImageUploadField"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={0}
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
            />
          </svg>
           
          Adicionar foto
        </span>
      </label>
      <span
        className="MuiTypography-root MuiTypography-srOnly"
      >
        <input
          accept="image/*"
          capture={null}
          data-testid="carFrontImageUploadField"
          id="carFrontImageUploadField"
          name="carFrontImage"
          onChange={[Function]}
          required={true}
          tabIndex={-1}
          type="file"
        />
      </span>
    </p>
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="previewFade"
      style={
        Object {
          "opacity": 0,
          "visibility": "hidden",
        }
      }
    >
      <figure
        className="MuiBox-root MuiBox-root-87"
      >
        <div
          className="MuiTypography-root MuiTypography-srOnly"
        >
          <figcaption
            className="MuiBox-root MuiBox-root-88"
          >
            Prévia da frente do veículo
          </figcaption>
        </div>
        <img
          alt="Prévia da frente do veículo"
          className="makeStyles-root-89"
          loading="lazy"
          src={null}
        />
      </figure>
    </div>
  </fieldset>
  <progress />
</form>
`;
